service/frontend/workflow_handler.go

```diff
@@ -467,6 +467,99 @@ func (wh *WorkflowHandler) unaliasedSearchAttributesFrom(
 	return sa, nil
 }
 
+func (wh *WorkflowHandler) ExecuteMultiOperation(
+	ctx context.Context,
+	request *workflowservice.ExecuteMultiOperationRequest,
+) (_ *workflowservice.ExecuteMultiOperationResponse, retError error) {
+	defer log.CapturePanic(wh.logger, &retError)
+
+	if request == nil {
+		return nil, errRequestNotSet
+	}
+
+	if !wh.config.EnableExecuteMultiOperation(request.Namespace) {
+		return nil, errMultiOperationAPINotAllowed
+	}
+
+	// TODO: validation
+
+	namespaceName := namespace.Name(request.GetNamespace())
+	namespaceID, err := wh.namespaceRegistry.GetNamespaceID(namespaceName)
+	if err != nil {
+		return nil, err
+	}
+
+	workflowId := request.Operations[0].GetStartWorkflow().WorkflowId
+	historyReq := &historyservice.ExecuteMultiOperationRequest{
+		NamespaceId: namespaceID.String(),
+		WorkflowId:  workflowId,
+		Operations:  make([]*historyservice.ExecuteMultiOperationRequest_Operation, len(request.Operations)),
+	}
+	for i, op := range request.Operations {
+		var opReq *historyservice.ExecuteMultiOperationRequest_Operation
+		if startReq := op.GetStartWorkflow(); startReq != nil {
```

Comment:
nit: would you mind pulling these switches out of the core logic and into a helper? This function will become unwieldy as we add operations

Reply:
I'm not sure what you have in mind. The two switches (one request and the other one for response) are operating on different types, so extracting a meaningful helper would be difficult.

Reply:
I just meant pull the switch itself out so that the higher-level code isn't obscured by a growing number of switch cases